Jenkinsfile

```
@@ -42,30 +36,126 @@ def publishReport(String label) {
+  def mdbRealmImage = docker.image("${env.DOCKER_REGISTRY}/ci/mongodb-realm-images:${version}")
+  def stitchCliImage = docker.image("${env.DOCKER_REGISTRY}/ci/stitch-cli:190")
+  docker.withRegistry("https://${env.DOCKER_REGISTRY}", "ecr:eu-west-1:aws-ci-user") {
+    mdbRealmImage.pull()
+    stitchCliImage.pull()
+  }
+  // run image, get IP
+  withDockerNetwork { network ->
+    mdbRealmImage.withRun("--network=${network} --network-alias=mongodb-realm") {
+      stitchCliImage.inside("--network=${network}") {
+        sh '''
+          echo "Waiting for Stitch to start"
+          while ! curl --output /dev/null --silent --head --fail http://mongodb-realm:9090; do
+            sleep 1 && echo -n .;
+          done;
+        '''
+
+        def access_token = sh(
+          script: 'curl --request POST --header "Content-Type: application/json" --data \'{ "username":"unique_user@domain.com", "password":"password" }\' http://mongodb-realm:9090/api/admin/v3.0/auth/providers/local-userpass/login -s | jq ".access_token" -r',
+          returnStdout: true
+        ).trim()
+
+        def group_id = sh(
+          script: "curl --header 'Authorization: Bearer $access_token' http://mongodb-realm:9090/api/admin/v3.0/auth/profile -s | jq '.roles[0].group_id' -r",
+          returnStdout: true
+        ).trim()
+
+        sh """
+          pwd && ls && ls ${configPath}
+          /usr/bin/stitch-cli login --config-path=${configPath}/stitch-config --base-url=http://mongodb-realm:9090 --auth-provider=local-userpass --username=unique_user@domain.com --password=password
+          /usr/bin/stitch-cli import --config-path=${configPath}/stitch-config --base-url=http://mongodb-realm:9090 --app-name auth-integration-tests --path=${configPath} --project-id $group_id -y --strategy replace
+        """
+      }
+      block(network)
+    }
+  }
 }
 
+
 def doDockerBuild(String flavor, Boolean withCoverage, Boolean enableSync, String sanitizerFlags = "") {
   def sync = enableSync ? "sync" : ""
   def label = "${flavor}${enableSync ? '-sync' : ''}"
+  def buildDir = "build"
+  def cmakeFlags = ""
+  if (withCoverage) {
+    cmakeFlags += "-DCMAKE_BUILD_TYPE=Coverage "
+  } else {
+    cmakeFlags += "-DCMAKE_BUILD_TYPE=RelWithDebInfo "
+  }
+  cmakeFlags += sanitizerFlags;
 
   return {
-    node('docker') {
-      getSourceArchive()
-      def image = buildDockerEnv("ci/realm-object-store:${flavor}")
-      sshagent(['realm-ci-ssh']) {
-        image.inside("-v /etc/passwd:/etc/passwd:ro -v ${env.HOME}:${env.HOME} -v ${env.SSH_AUTH_SOCK}:${env.SSH_AUTH_SOCK} -e HOME=${env.HOME}") {
-          if(withCoverage) {
-            sh "rm -rf coverage.build ${label}.build && ./workflow/test_coverage.sh ${sync} && mv coverage.build ${label}.build"
-          } else {
-            sh "./workflow/build.sh ${flavor} ${sync} ${sanitizerFlags}"
+    nodeWithSources('docker') {
+      def image = docker.build("ci/realm-object-store:${flavor}")
```

```suggestion
      def image = buildDockerEnv('ci/realm-object-store:linux', push: env.BRANCH_NAME == 'master')
```
This will use https://github.com/realm/ci/blob/master/vars/buildDockerEnv.groovy. You can add a `push: true` to the arguments to have the image be cached on our AWS docker registry.